Review of @assert docs (#2292)
cap-js/docs: guides/providing-services.md

### `@assert` <Beta/>

> Annotate an element with `@assert` to define CXL expressions that validate data after you write it to the database but before the system commits it. If validation fails, the expression returns a `String` that indicates an error to the runtime. If validation passes, the expression returns `null`. 

Reviewer: ```suggestion
The `@assert` annotation lets you define a CXL expression that is used to validate data _after_ it's been written to the database but _before_ it's committed. If validation fails, the expression returns a `String` that indicates an error to the runtime. If validation passes, the expression returns `null`. 
```

#### Error Messages and Message Targets

> If validation fails, the transaction is rolled back with an exception. If you use [Fiori draft state messages](../advanced/fiori#validating-drafts), the error is persisted. The error targets the annotated element, which is then highlighted on the Fiori UI.

Author: Alright, we'll find out through issues/incidents if we need to explain the "more complicated" stuff as well. Thanks!
